Objects/unionobject.c

```diff
@@ -242,7 +295,35 @@ _Py_union_type_or(PyObject* self, PyObject* other)
         Py_RETURN_NOTIMPLEMENTED;
     }
 
-    PyObject *tuple = PyTuple_Pack(2, self, other);
+    if (self == Py_None) {
+        self = (PyObject *)&_PyNone_Type;
+    }
+    if (other == Py_None) {
+        other = (PyObject *)&_PyNone_Type;
+    }
+
+    PyObject *tuple;
+
+    if (_PyUnion_Check(self) && _PyUnion_Check(other)) {
+        tuple = merge_union_and_union(self, other);
+    } else if (_PyUnion_Check(self)) {
+        tuple = merge_union_and_obj(self, other);
+    } else if (_PyUnion_Check(other)) {
+        tuple = merge_obj_and_union(self, other);
+    } else {
```

Comment:
Is this code reachable?

Reply:
Yes, it handles cases when `self` and `other` aren't `UnionType`.
For instance:
```py
int | float
```